nameserver/nameserver.go

```diff
@@ -87,8 +89,8 @@ func handleRequest(ctx context.Context, w dns.ResponseWriter, request *dns.Msg)
 		return reply(nsutil.Refused("unsupported qclass"))
 	}
 
-	// Handle request for localhost.
-	if strings.HasSuffix(q.FQDN, "localhost.") {
+	// Handle request for localhost and the hostname.
+	if strings.HasSuffix(q.FQDN, "localhost.") || q.FQDN == hostname {
 		tracer.Tracef("nameserver: returning localhost records")
 		return reply(nsutil.Localhost())
 	}
```

Comment:
The `hostname` is often resolved to a different IP address like `´127.0.1.1` or a static IP assigned to the primary interface. I guess if we want to handle that correctly we should finally start parsing `/etc/hosts` on linux and use that.

Reply:
Afaict, the `127.0.1.1` is used by the OS to do a IP -> hostname mapping, not necessarily for resolving.
I don't think this should be a problem, and if it does, we'll handle it then.
Right now, it's an improvement because some tools (eg. sudo) do want to resolve the hostname via DNS.